nullifier-merkle-tree/src/lib.rs

```diff
@@ -0,0 +1,481 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum NullifierMerkleTreeError {
+    #[error("Low nullifier not found")]
+    LowNullifierNotFound,
+}
+
+#[derive(Default, Debug, PartialEq)]
+pub struct Node {
+    value: [u8; 32],
+    next_index: usize,
+    next_value: [u8; 32],
+}
+
+pub struct NullifierMerkleTree<const HEIGHT: usize> {
+    pub subtrees: [Node; HEIGHT],
```

Comment:
Isn’t it just the vector of nodes not the subtrees?
Also the current root is missing in this implementation. I suppose we would want to extend the insert function to update the root as well?

Reply:
Good points.

I renamed `subtrees` to `nodes` and added the roots already.